### ISSUE TITLE
[Feat] Add a feature to save books to your favorites

### ISSUE THREAD

**Issue by gbowne1** [gbowne1/codebooker#30]
Some users may want to save books to their favorites.

May even have a side feature for users to share their favorite books with others.

I would suggest using a heart icon to:

1) add to favorite books list
2) show favorited books

Do we need to build a UI feature that will allow users to see their and/or possibly other users favorite books? My thoughts are yes.

**Comment by Gourav2609** [comment]
hello @gbowne1 ....can u please assign this to me......i think i will be able to this

**Comment by gbowne1** [comment]
ok @Gourav2609 I will assign you.

**Comment by Gourav2609** [comment]
Thanks @gbowne1 for giving this opportunity....this is going to be my first time experience....thanks a lot.....

**Comment by gbowne1** [comment]
You're welcome @Gourav2609.

**Comment by Gourav2609** [comment]
@gbowne1 
![image](https://github.com/gbowne1/codebooker/assets/115522947/605b736b-385f-4b63-8137-5602ee281fc8)
can u please help me here ...i have tried it 

i have created the .env file 
![image](https://github.com/gbowne1/codebooker/assets/115522947/8ea90488-6c20-4146-a550-1ab80abfe488)

**Comment by gbowne1** [comment]
@LOGESH-B

Any ideas? I was not able to replicate this error.

**Comment by gbowne1** [comment]
@Gourav2609 

I will say that the connection string in the MONGO_DB string should be:

MONGO_DB=mongodb://username:password@localhost:27017/codebooker

replace username and password with your local database password and make sure there is an : in between them

**Comment by gbowne1** [comment]
I also will say that it won't work without starting the `mongod` server first and then using either `mongo` or `mongosh` to create the `codebooker` database by doing `use codebooker` in either mongo or mongosh.  If you use MongoDB compass, MongoDB atlas or the MongoDB extension in VSCode the workflow will be a bit different.  You can use whatever database name you want in place of `codebooker`.

**Comment by LOGESH-B** [comment]
Hii @Gourav2609 
I think insertOne is not supported by mongoose.....insertOne is native to MongoDb

**Comment by LOGESH-B** [comment]
Hii @Gourav2609 
Can you share your setup.js file.....because in the code there is no insertOne function

**Comment by gbowne1** [comment]
Well, yes and no.  

The insertOne() method is not a native method of Mongoose. It is a method provided by the MongoDB Node.js driver

In Mongoose, you can use the create() method to insert a document into a collection

If you want to use the insertOne() method from the MongoDB Node.js driver directly, you can access the underlying collection object using Model.collection and then call the insertOne() method on it.

**Comment by Gourav2609** [comment]
> I also will say that it won't work without starting the `mongod` server first and then using either `mongo` or `mongosh` to create the `codebooker` database by doing `use codebooker` in either mongo or mongosh. If you use MongoDB compass, MongoDB atlas or the MongoDB extension in VSCode the workflow will be a bit different. You can use whatever database name you want in place of `codebooker`.

actually i was using mongoDB campass

**Comment by gbowne1** [comment]
@Gourav2609 

I don't think you'll need to run setup.js if you use MongoDB compass. All setup.js does is create a test user.

That can be done like this in Compass, according to some documentation.

To create a test user using MongoDB Compass, you don't need to run the setup.js file mentioned in your previous question. Instead, you can follow these general steps:

- Open MongoDB Compass and connect to your MongoDB server instance. You can refer to the MongoDB Compass documentation for detailed instructions on how to connect.

- Once connected, navigate to the database where you want to create the test user. 

- In the database view, look for an option to create a new user. The exact steps may vary depending on the version of MongoDB Compass you are using. Typically, you would find an option like "Create Database User" or "Add User". Click on that option.

- Enter the necessary details for the test user, such as username, password and email.

Save the user configuration. MongoDB Compass will create the user in the specified authentication database.

By following these steps, you should be able to create a test user in your MongoDB database using MongoDB Compass without the need to run the setup.js file.

**Comment by Gourav2609** [comment]
thanks @gbowne1 I will try it and get back to you if I face any issues..

**Comment by gbowne1** [comment]
yeah, Please do let @LOGESH-B and me know about any issues you have. He uses Atlas.  I use mongod and mongosh locally in the console/terminal.  I've never used Compass or Atlas so I don't know how that worflow goes, just did some reading to find out those steps.

**Comment by Gourav2609** [comment]
Hello @gbowne1 and @LOGESH-B ,
### My Progress

**1. Ran MongoDB server** 
![image](https://github.com/gbowne1/codebooker/assets/115522947/ccbd666e-5371-4f96-8ec2-375d213d14bd)

**2. **Ran mongosh command****  
![image](https://github.com/gbowne1/codebooker/assets/115522947/a82042e8-ec46-467b-bbac-9174b2fc870f)

**3.Ran setup.js**
![image](https://github.com/gbowne1/codebooker/assets/115522947/d5683a0c-49a2-4c7d-ae84-e0080509d484)

**4. Ran index.js**
![image](https://github.com/gbowne1/codebooker/assets/115522947/e00f28de-0ca5-4592-bbc7-abdc02263e1c)

**5. Copied username and password from users and pasted it in React App**
![image](https://github.com/gbowne1/codebooker/assets/115522947/64332f15-d42c-4e57-9ed1-4d948fd43b4a)
![image](https://github.com/gbowne1/codebooker/assets/115522947/2416a565-11a2-49f9-b5ee-39a1fb5d408d)


### Still getting this error
![image](https://github.com/gbowne1/codebooker/assets/115522947/f964fb9d-948e-44ea-b8e5-c0a8ef803405)

**Comment by gbowne1** [comment]
that is because you copied the hash of the password.  the actual password is `test` iirc.

**Comment by Gourav2609** [comment]
@gbowne1 thanks for the correction
now i was able to open it
i was thinking since i learned a lot during setup for the project ........how about me contributing to a proper documentation about setup of the project

**Comment by gbowne1** [comment]
Thanks  @Gourav2609!  Glad that helped out. Yes the test user login information is in the User model/schema.

Yes, and  you're always always welcome to work on anything you feel like working on.  There is an issue for adding MongoDB Compass/Atlas #137 

Any documentation go in /docs/ and also linked in the README's.  

They will eventually get cross posted to the Wiki.. there is an issue for that #141

**Comment by Gourav2609** [comment]
@gbowne1 

### Getting Started With UI
![image](https://github.com/gbowne1/codebooker/assets/115522947/27dcf609-c66e-4e7b-b752-2587678c999f)

### Thinking

i was thinking of creating a similar column with a name Favourite

initially every book contains 
![image](https://github.com/gbowne1/codebooker/assets/115522947/0eee7829-8283-4535-96aa-d6314e02e551)
whenever a book is added to favorites 
the icon changes to 
![image](https://github.com/gbowne1/codebooker/assets/115522947/2b3c3556-5090-4bd7-a45f-103c98d18687)

### on cliking favourites

![image](https://github.com/gbowne1/codebooker/assets/115522947/562b6b5c-4d26-42cd-9922-90dc94e13910)

using same color pallets and displaying them

how do you think this would be

**Comment by gbowne1** [comment]
@Gourav2609 I love the idea about changing the icon filled state when the favorite gets added.

Would clicking on Favorites in the dropdown/context menu there show all the favorited items?

We would have to figure out where to put a heart / favorite icon for actually adding the item.

I think that's great

**Comment by Gourav2609** [comment]
@gbowne1 
> Would clicking on Favorites in the dropdown/context menu there show all the favorited items?
> 
when the user navigates to favourite through profile then the user is navigated to a different page  where he can view only his favourite books .

> We would have to figure out where to put a heart / favorite icon for actually adding the item.
> 

how about before review and after year columns

**Comment by gbowne1** [comment]
sounds like a great idea.

**Comment by gbowne1** [comment]
@Gourav2609 it's been almost a month, have you been able to get through this issue?  Just checking in to see if you have made any progress.

**Comment by Gourav2609** [comment]
hello @gbowne1 ....sorry not able to update you about the progress...i just got  a little busy with my college things...will update soon..

**Comment by gbowne1** [comment]
Ok, let me know. Am just checking in to see if you have made any progress.

**Comment by akash0708** [comment]
@gbowne1 I would like to give this a go.

**Comment by gbowne1** [comment]
Ok @akash0708 I will assign you

**Comment by akash0708** [comment]
@gbowne1 I have been facing this error since yesterday,

![image](https://github.com/gbowne1/codebooker/assets/110753356/eec72187-b43d-40f8-96c1-a15ed621e662)

I've run script.js, created the test user. But I'm not being able to log in

**Comment by gbowne1** [comment]
the test user credentials are:

username: test
email: test@gmail.com
password test

Also: https://github.com/gbowne1/codebooker/blob/master/docs/DEVSETUP.md has more information, as well as the rest of this thread.  If it still persists.. there might be other issues.. let us know.  

@BlackBond06 @LOGESH-B

**Comment by LOGESH-B** [comment]
Hii @akash0708 
Feel free to ask if you encountered any issue.🤝
